Makefile

```diff
@@ -126,6 +129,11 @@ test: ## Run tests.
 test-verbose: ## Run tests with verbose settings.
 	TEST_ARGS="$(TEST_ARGS) -v" $(MAKE) test
 
+.PHONY: test-junit
+test-junit: $(GO_JUNIT_REPORT) ## Run tests with verbose setting and generate a junit report.
+	($(MAKE) test-verbose; echo $$? > $(ARTIFACTS)/junit.exitcode) 2>&1 | tee /dev/stderr | $(GO_JUNIT_REPORT) > $(ARTIFACTS)/junit.xml
+	exit $$(cat $(ARTIFACTS)/junit.exitcode)
+
```

Comment:
This looks quite hacky but it was the simplest way I could achieve the following:
* print the output of go test to the terminal (`tee /dev/stderr`)
* always generate a junit.xml, especially when go test fails
* propagate the exit code, so the make target still fails when go test fails

Reply:
In addition: now basically the same implementation as k/k: https://github.com/kubernetes/kubernetes/blob/f981628ce897a96a8a08a2a21c4d6bdb41601fbd/hack/make-rules/test.sh#L249-L256

We now have full package paths both in the Deck junit lens and in testgrid